actionpack/lib/action_dispatch/http/response.rb

```diff
@@ -221,6 +221,9 @@ def sent?;      synchronize { @sent };      end
 
     # Sets the HTTP status code.
     def status=(status)
+      raise ArgumentError, "#{status} is not a valid response code identifier. See " \
```

Comment:
* Do you mind using `status.inspect` so it would print out `:i_am_a_teapot is not a valid ...` prefixed with colon?
* Minor thing, but given the `unless` is 2 lines below the start of this code I feel like using a normal `if` block would improve the readability and reduce the surprise. What do you think?

  ```ruby
  if valid_http_status?(status)
    @status = Rack::Utils.status_code(status)
  else
    raise ArgumentError, "#{status.inspect} is not a valid response code identifier. See " \
      "http://guides.rubyonrails.org/layouts_and_rendering.html#the-status-option " \
      "for valid values" unless valid_http_status?(status)
  end

Reply:
:+1: for the `if/else` and the inspect